### PR TITLE
Fullscreen Resolution Slider

### LINKED ISSUE #2692: Add option to change game resolution
### Request Description

In vanilla Minecraft, you can change the resolution/frequency. This is important because it defaults to 1080p60, I want to change it to 1080p75 (the frequency of my monitor). However when sodium is installed, this optioin seems to disapear.

### PR DESCRIPTION
**Pull request by douira**
As a compromise, this adds a fullscreen resolution slider but truncates the label to fit into one line. The design can be refined later, but at least users won't have to go to the vanilla settings for this now.
Thanks to @MeeniMc for their previous work in https://github.com/CaffeineMC/sodium-fabric/pull/2296 on adding the option in the first place. I added the two-line functionality and changed the style.
related: https://github.com/CaffeineMC/sodium-fabric/issues/46
Closes https://github.com/CaffeineMC/sodium-fabric/issues/2692

### CODE REVIEW

**Comment by MeeniMc** [comment]
It's a good idea but it doesn't work with AUTO gui scale, there is no vertical scrolling and the extra line renders outside the screen now.

**Comment by douira** [comment]
That's a good point, do you know where one makes the control element notify the GUI system of it's correct size? Is there a place where it calculates the total size of the GUI for determining the scale?

**Comment by douira** [comment]
I've updated it to be only one line but truncate the width of the label dynamically based on the slider's value width.

**Comment by MeeniMc** [comment]
LGTM

**Comment by douira** [comment]
resolved conflicts and brought the branch up to speed with dev

**Comment by douira** [comment]
Closes https://github.com/CaffeineMC/sodium-fabric/issues/2692

File: common/src/main/java/net/caffeinemc/mods/sodium/client/gui/options/control/ControlElement.java [hunk 1]
```diff
@@ -20,18 +20,50 @@ public ControlElement(Option<T> option, Dim2i dim) {
...
 
-        if ((this.hovered || this.isFocused()) && this.font.width(name) > (this.dim.width() - this.option.getControl().getMaxWidth())) {
-            name = name.substring(0, Math.min(name.length(), 10)) + "...";
+        // add the star suffix before truncation to prevent it from overlapping with the label text
+        if (this.option.isAvailable() && this.option.hasChanged()) {
+            name = name + " *";
         }
 
+        // on focus or hover truncate the label to never overlap with the control's content
+        if (this.hovered || this.isFocused()) {
```
Comment (jellysquid3) [inline common/src/main/java/net/caffeinemc/mods/sodium/client/gui/options/control/ControlElement.java:38]: This should really be a separate utility function.

**Review by jellysquid3 — APPROVED** [review]

File: common/src/main/java/net/caffeinemc/mods/sodium/client/gui/options/control/ControlElement.java (hunk 1 above)
Comment (douira) [reply common/src/main/java/net/caffeinemc/mods/sodium/client/gui/options/control/ControlElement.java:38]: good point

Comment (douira) [reply common/src/main/java/net/caffeinemc/mods/sodium/client/gui/options/control/ControlElement.java:38]: done

**Comment by jellysquid3** [comment]
This is fine as a temporary solution for now. But I wouldn't want to keep around multi-line controls once we rework the video setting screen.

**Comment by douira** [comment]
This PR doesn't add it as a multiline option (anymore), I replaced it with label truncation.

**Comment by douira** [comment]
My bad for not updating the description

**Pull request merged**